### ISSUE TITLE
Inline importing function from same module causes `unresolved-import` error

### ISSUE THREAD

**Issue by sid-kap** [astral-sh/ty#2596]
### Summary

Importing the same module inline works:
```py
def foo() -> int:
    return 1

def bar() -> int:
    import main
    return main.foo()
```
https://play.ty.dev/5e93575f-83a5-4186-bf04-0babd27364b1

But importing a function from the same module triggers an `unresolved-import` error:
```py
def foo() -> int:
    return 1

def bar() -> int:
    from main import foo
    return foo()
```
```
Module `main` has no member `foo` (unresolved-import) [Ln 5, Col 22]
```
https://play.ty.dev/83b6a067-c32d-432e-a7da-f0eea1bdd6e2

I believe the second example should pass typechecking?

### Version

ty 0.0.12 (Homebrew 2026-01-14)

**Comment by carljm** [comment]
Yes, I think it should -- thanks for the report!

**Comment by sid-kap** [comment]
Thank you so much!!
